libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3046,6 +3046,12 @@ bool TypeChecker::visit(Identifier const& _identifier)
 	return false;
 }
 
+void TypeChecker::endVisit(UserDefinedTypeName const& _userDefinedTypeName)
+{
+	if (!_userDefinedTypeName.annotation().type)
```

Comment:
When would this be the case?

Reply:
Usually the `DeclarationTypeChecker::visit(UserDefinedTypeName..)` would assign the type, but we can't do that for modifiers because they require information only available in this stage so for the modifiers we have to do it here.